server/src/main/java/org/opensearch/rest/action/admin/cluster/RestWlmStatsAction.java

```diff
@@ -53,6 +65,105 @@ protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient cli
+                        }
+
+                        request.params().put("v", "true");
+                        return RestTable.buildResponse(paginatedTable, channel);
+                    }
+                }
+            );
+        }
+
         return channel -> client.admin().cluster().wlmStats(wlmStatsRequest, new RestActions.NodesResponseRestListener<>(channel));
     }
+
+    /**
+     * Builds a tabular response with '|' column separators.
+     */
+    private Table buildTable(List<WlmStats> paginatedStats, WlmPaginationStrategy paginationStrategy) {
+        Table table = new Table();
+        table.startHeaders();
+        table.addCell("NODE_ID", "desc:Node ID");
+        table.addCell("|");
+        table.addCell("QUERY_GROUP_ID", "desc:Query Group");
+        table.addCell("|");
+        table.addCell("TOTAL_COMPLETIONS", "desc:Total Completed Queries");
+        table.addCell("|");
+        table.addCell("TOTAL_REJECTIONS", "desc:Total Rejected Queries");
+        table.addCell("|");
+        table.addCell("TOTAL_CANCELLATIONS", "desc:Total Canceled Queries");
+        table.addCell("|");
+        table.addCell("CPU_USAGE", "desc:CPU Usage");
+        table.addCell("|");
+        table.addCell("MEMORY_USAGE", "desc:Memory Usage");
+        table.endHeaders();
+
+        for (WlmStats wlmStats : paginatedStats) {
+            QueryGroupStats queryGroupStats = wlmStats.getQueryGroupStats();
+
+            for (Map.Entry<String, QueryGroupStats.QueryGroupStatsHolder> entry : queryGroupStats.getStats().entrySet()) {
+                String queryGroupId = entry.getKey();
+                QueryGroupStats.QueryGroupStatsHolder statsHolder = entry.getValue();
+
+                table.startRow();
+                table.addCell(wlmStats.getNode().getId());
+                table.addCell("|");
+                table.addCell(queryGroupId);
+                table.addCell("|");
+                table.addCell(statsHolder.getCompletions());
+                table.addCell("|");
+                table.addCell(statsHolder.getRejections());
+                table.addCell("|");
+                table.addCell(statsHolder.getCancellations());
+                table.addCell("|");
+
+                QueryGroupStats.ResourceStats cpuStats = statsHolder.getResourceStats().get(ResourceType.CPU);
+                QueryGroupStats.ResourceStats memoryStats = statsHolder.getResourceStats().get(ResourceType.MEMORY);
+
+                table.addCell(cpuStats != null ? cpuStats.getCurrentUsage() : 0);
+                table.addCell("|");
+                table.addCell(memoryStats != null ? memoryStats.getCurrentUsage() : 0);
+                table.endRow();
+            }
+        }
+
+        if (paginationStrategy.getNextToken() == null) {
+            table.startRow();
+            table.addCell("No more pages available");
+            for (int i = 1; i < 13; i++) {
+                table.addCell("-");
+            }
+            table.endRow();
```

Comment:
Should this be conditional ? Additionally lets refactor a method for this footer .